test/integration/hotreload_test.lua

```diff
@@ -0,0 +1,98 @@
+local fio = require('fio')
```

Comment:
What property of expirationd you verify here? I don't understand how all this code is relevant to expirationd.

Reply:
There should be a well defined hypothesis about a class of problems we can catch this way (plus understanding why we can't do it in a more obvious and simple way) to actually add and support this testing code within the expirationd repository.

Reply:
Please, take me right: I encourage using of smart testing techniques, where usual unit / functional testing are not powerful enough. Say, when a functional testing is able to catch particular set of highly specialized problems, but well implemented integration testing is able to catch almost any problem of given class. Or, say, the typical situation, when a functional testing unable to catch a real world problem, which only appears with data from a previous version of a module / an application.

I understand what cartridge/hotreload.lua doing (as well as moonlibs/module-reload). But I still don't catch the idea of this test. The questions that I ask for myself are:

* What is the class of problems it should catch?
* Whether it is beyond the class of problems that will be catched by the following code:

  ```lua
  local expirationd = require('expirationd')

  <..create a space..>
  <..fill the space..>
  expirationd.start(<...>)
  <..check all tuples are expired..>

  expirationd.kill(<...>)
  <..fill the space..>
  <..check that tuples are not expiring..>

  package.loaded.expirationd = nil

  expirationd = require('expirationd')
  expirationd.start(<...>)
  <..check all tuples are expired..>
  ```
* Even more, whether it is beyoud problems we able to catch this way:

  ```lua
  require('expirationd')
  package.loaded.expirationd = nil
  require('expirationd')
  ```

Reply:
in this case, @no1seman asked to make a test, which will clearly show that when call require("cartridge.roles").reload(), correct processing of the expirationd module is possible